vendor/github.com/kubeflow/tf-operator/pkg/control/service_control.go

```diff
@@ -200,7 +201,9 @@ func (f *FakeServiceControl) CreateServicesWithControllerRef(namespace string, s
 func (f *FakeServiceControl) DeleteService(namespace string, serviceID string, object runtime.Object) error {
 	f.Lock()
 	defer f.Unlock()
-	f.DeleteServiceName = append(f.DeleteServiceName, serviceID)
+	if strings.Contains(serviceID, "master"){
```

Comment:
It's a little hack. I am wondering if we need it

Reply:
Because function "deletePodsAndServices" will delete all the services of pod, so if we don not change the function  "deletePodsAndServices"(only delete the master service), maybe it's difficult to pass the test case without change the function DeleteService(FakeServiceControl)


```
func (pc *PyTorchController) deletePodsAndServices(job *pyv1.PyTorchJob, pods []*v1.Pod) error {
	if len(pods) == 0 {
		return nil
	}

	// Delete nothing when the cleanPodPolicy is None or Running.
	if *job.Spec.CleanPodPolicy == common.CleanPodPolicyNone ||
		*job.Spec.CleanPodPolicy == common.CleanPodPolicyRunning {
		return nil
	}

	for _, pod := range pods {
		if err := pc.PodControl.DeletePod(pod.Namespace, pod.Name, job); err != nil {
			return err
		}
		// Pod and service have the same name, thus the service could be deleted using pod's name.
		if err := pc.ServiceControl.DeleteService(pod.Namespace, pod.Name, job); err != nil {
			return err
		}
	}
	return nil
}
```

Reply:
Gotcha. SGTM. I think it is a potential issue in our code base. We can have another PR to avoid deleting unexisting worker services.

Reply:
yes, it will be better to avoid deleting unexisting worker services.

Reply:
Can you take this cleanup in a separate PR?

Reply:
@leileiwan can you create an issue to track this

Reply:
@johnugeorge ,ok